packages/patternfly-3/patternfly-react/src/components/DateTimePicker/TimePicker.js

```diff
@@ -0,0 +1,103 @@
+import { FormControl, InputGroup, Icon, OverlayTrigger, Popover } from '../../';
+import TimeInput from './TimeComponents/TimeInput';
+
+class TimePicker extends React.Component {
+  getDateFromTime = time => {
+    if (Date.parse(time)) {
+      return new Date(time);
+    }
+    return new Date(`1/1/1 ${time}`);
+  };
+
+  formatDate = time => {
+    const { locale } = this.props;
+    const options = { hour: '2-digit', minute: '2-digit' };
+    if (Date.parse(time)) {
+      const parsedValue = new Date(time);
+      return parsedValue.toLocaleString(locale, options);
+    } else if (Date.parse(`1/1/1 ${time}`)) {
+      const parsedValue = new Date(`1/1/1 ${time}`);
+      return parsedValue.toLocaleString(locale, options);
+    }
+    return time;
+  };
+  state = {
+    value: this.getDateFromTime(this.props.value),
+    tmpValue: this.formatDate(this.props.value),
+    hiddenValue: this.props.hiddenValue
+  };
+  setSelected = time => {
+    let newTime = new Date(this.state.value);
+    if (Date.parse(time)) {
+      newTime = new Date(time);
+    } else if (Date.parse(`1/1/1 ${time}`)) {
+      newTime = new Date(`1/1/1 ${time}`);
+    }
+    this.setState({ value: newTime, tmpValue: this.formatDate(newTime) });
+  };
+  render() {
+    const { value, hiddenValue } = this.state;
+    const { locale, placement, id } = this.props;
+    const popover = (
+      <Popover
+        id={id}
+        className="bootstrap-datetimepicker-widget date-time-picker-pf bootstrap-timepicker-widget  dropdown-menu"
+      >
+        <ul className="list-unstyled">
+          <li className="picker-switch accordion-toggle">
+            <table className="table-condensed">
+              <tbody>
+                <tr />
+              </tbody>
+            </table>
+          </li>
+          <li>
+            <TimeInput time={value} setSelected={this.setSelected} locale={locale} />
+          </li>
+        </ul>
+      </Popover>
+    );
+
+    return (
+      <InputGroup className="input-group date-time-picker-pf time-picker-pf">
+        <FormControl
+          aria-label="time-picker-input"
+          type="text"
+          value={hiddenValue ? '' : this.state.tmpValue}
+          onChange={e => this.setState({ tmpValue: e.target.value })}
+          onBlur={e => this.setSelected(e.target.value)}
+        />
+        <OverlayTrigger trigger="click" placement={placement} overlay={popover} rootClose>
+          <InputGroup.Addon className="picker-pf">
+            <Icon type="fa" name="clock-o" />
+          </InputGroup.Addon>
+        </OverlayTrigger>
+        <InputGroup.Addon className="clear-button">
+          <Icon
+            type="pf"
+            name={hiddenValue ? 'edit' : 'close'}
+            onClick={() => this.setState({ hiddenValue: !hiddenValue })}
+          />
+        </InputGroup.Addon>
+      </InputGroup>
+    );
+  }
+}
+
+TimePicker.propTypes = {
+  value: PropTypes.oneOfType([PropTypes.instanceOf(Date), PropTypes.string]),
```

Comment:
can you add documentation above all of the propTypes so it would be visible in the storybook ?